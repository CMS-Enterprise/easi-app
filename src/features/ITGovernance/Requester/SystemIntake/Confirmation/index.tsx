import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link as ReactRouterLink, useParams } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  ButtonGroup,
  GridContainer,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading
} from '@trussworks/react-uswds';

import ExternalLinkAndModal from 'components/ExternalLinkAndModal';
import PageHeading from 'components/PageHeading';

const Confirmation = ({
  submissionSuccess
}: {
  submissionSuccess: boolean;
}) => {
  const { systemId } = useParams<{ systemId: string }>();
  const { t } = useTranslation('intake');

  return (
    <>
      <div
        className={`margin-bottom-8  padding-bottom-6 ${submissionSuccess ? 'bg-success-lighter' : 'bg-error-lighter'}`}
      >
        <GridContainer>
          <BreadcrumbBar
            variant="wrap"
            className={`margin-bottom-6 ${submissionSuccess ? 'bg-success-lighter' : 'bg-error-lighter'}`}
          >
            <Breadcrumb>
              <BreadcrumbLink asCustom={ReactRouterLink} to="/">
                <span>{t('taskList:navigation.home')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb>
              <BreadcrumbLink
                asCustom={ReactRouterLink}
                to={`/governance-task-list/${systemId}`}
              >
                <span>{t('taskList:navigation.governanceTaskList')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb current>
              {t('taskList:navigation.intakeRequest')}
            </Breadcrumb>
          </BreadcrumbBar>

          <PageHeading className="margin-top-0 margin-bottom-1">
            {submissionSuccess
              ? t('submission.success.heading')
              : t('submission.error.heading')}
          </PageHeading>
          <p className="font-body-lg line-height-body-5 margin-top-0 margin-bottom-5 text-light">
            {submissionSuccess
              ? t('submission.success.description')
              : t('submission.error.description')}
          </p>
          <ButtonGroup>
            {!submissionSuccess && (
              <ReactRouterLink
                to={`/system/${systemId}/contact-details`}
                className="usa-button"
              >
                {t('submission.error.backToIntakeRequest')}
              </ReactRouterLink>
            )}
            <ReactRouterLink
              to={`/governance-task-list/${systemId}`}
              className={`usa-button ${!submissionSuccess ? 'usa-button--outline' : ''}`}
            >
              {t('taskList:navigation.returnToTaskList')}
            </ReactRouterLink>
          </ButtonGroup>
        </GridContainer>
      </div>
      {submissionSuccess && (
        <div>
          <GridContainer>
            <SummaryBox className="grid-col-8 margin-top-0 margin-bottom-5">
              <SummaryBoxHeading headingLevel="h3" className="margin-bottom-2">
                {t('review.nextSteps.heading')}
              </SummaryBoxHeading>
              <SummaryBoxContent>
                {t('review.nextSteps.description')}
              </SummaryBoxContent>
            </SummaryBox>
            <p>{t('submission.success.learnMore')}</p>
            <ExternalLinkAndModal
              href={t('submission.success.sharepointLink.href')}
            >
              <span>{t('submission.success.sharepointLink.copy')}</span>
            </ExternalLinkAndModal>
          </GridContainer>
        </div>
      )}
    </>
  );
};

export default Confirmation;
