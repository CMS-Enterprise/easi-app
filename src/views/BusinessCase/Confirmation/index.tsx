import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { Trans, useTranslation } from 'react-i18next';

const Confirmation = () => {
  const { businessCaseId } = useParams();
  const { t } = useTranslation();
  return (
    <div className="margin-bottom-7">
      <div className="grid-container">
        <h1 className="font-heading-xl margin-top-4">
          {t('businessCase:submission.confirmation.heading')}
        </h1>
        <h2 className="margin-bottom-8 text-normal">
          {t('businessCase:submission.confirmation.subheading', {
            referenceId: businessCaseId
          })}
        </h2>
        <p className="margin-y-8">
          <Trans i18nKey="businessCase:submission.confirmation.improveEasi">
            helpUsImproveEasi
            <a
              href="https://www.surveymonkey.com/r/JNYSMZP"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Open EASi survey in a new tab"
            >
              tellUsWhatYouThink
            </a>
          </Trans>
        </p>
        <div>
          <Link to="/">
            <i className="fa fa-angle-left margin-x-05" aria-hidden />
            {t('businessCase:submission.confirmation.homeCta')}
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Confirmation;
