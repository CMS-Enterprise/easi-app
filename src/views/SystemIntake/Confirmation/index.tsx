import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useParams } from 'react-router-dom';
import { useFlags } from 'launchdarkly-react-client-sdk';

import { ImproveEasiSurvey } from 'components/Survey';

const Confirmation = () => {
  const { systemId } = useParams<{ systemId: string }>();
  const flags = useFlags();
  const { t } = useTranslation('intake');

  return (
    <div className="grid-container margin-bottom-7">
      <div>
        <h1 className="font-heading-xl margin-top-4">
          {t('submission.confirmation.heading')}
        </h1>
        <h2 className="margin-bottom-8 text-normal">
          {t('submission.confirmation.subheading', {
            referenceId: systemId
          })}
        </h2>
        <ImproveEasiSurvey />
        <div>
          {flags.taskListLite ? (
            <Link to={`/governance-task-list/${systemId}`}>
              <i className="fa fa-angle-left margin-x-05" aria-hidden />
              {t('submission.confirmation.taskListCta')}
            </Link>
          ) : (
            <Link to="/">
              <i className="fa fa-angle-left margin-x-05" aria-hidden />
              {t('submission.confirmation.homeCta')}
            </Link>
          )}
        </div>
      </div>
    </div>
  );
};

export default Confirmation;
