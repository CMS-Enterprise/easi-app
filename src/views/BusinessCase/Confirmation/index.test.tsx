import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Confirmation from './index';

describe('The Business Case Confirmation page', () => {
  it('renders without crashing', () => {
    shallow(
      <MemoryRouter>
        <Confirmation />
      </MemoryRouter>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Confirmation />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
