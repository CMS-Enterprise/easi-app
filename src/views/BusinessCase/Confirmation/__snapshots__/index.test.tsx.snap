// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Business Case Confirmation page matches the snapshot 1`] = `
<div
  className="margin-bottom-7"
>
  <div
    className="grid-container"
  >
    <h1
      className="font-heading-xl margin-top-4"
    >
      Your Business Case has been submitted
    </h1>
    <h2
      className="margin-bottom-8 text-normal"
    >
      Your reference ID is 
    </h2>
    <p
      className="margin-y-8"
    >
      Help us improve EASi. 
      <a
        aria-label="Open EASi survey in a new tab"
        href="https://www.surveymonkey.com/r/JNYSMZP"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tell us what you think of this service (opens in a new tab)
      </a>
    </p>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          aria-hidden={true}
          className="fa fa-angle-left margin-x-05"
        />
        Go back to EASi homepage
      </a>
    </div>
  </div>
</div>
`;
